flask-homework/views.py

```diff
@@ -172,6 +157,90 @@ def book_details(title):
     return render_template("books/book_details.html", **context), 200
 
 
+@app.get("/purchases")
+@is_user_logged_in
+def get_purchases():
+    query = db.select(User.first_name, User.last_name, Book.title, Book.author)\
+        .join(Purchase, User.id == Purchase.user_id)\
+        .join(Book, Book.id == Purchase.book_id)
+
+    purchases_render = db.session.execute(query)
+
+    context = {
+        "title": "Purchases_list",
+        "username": session["username"],
+        "purchases": purchases_render
+    }
+    return render_template("purchases/purchases.html", **context), 200
+
+
+@app.get("/purchases/<int:purchase_id>")
+@is_user_logged_in
+def purchase_details(purchase_id):
+    try:
+        purchase_id = int(purchase_id)
+    except ValueError:
+        abort(400, "Invalid purchase id")
+
+    purchase = None
+
+    query = db.select(Purchase)
+    purchases = db.session.execute(query).scalars()
```

Comment:
nit: I think it is better to use filter by id and join book and user to make only one query